### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by cs-utkarsh-surana**
#### UTCs:  

- [x] Connector installation verified.  
- [x] Connector logo verified.  
- [x] Actions and Playbook list verified.  
- [x] Playbook tags verified.  
- [x] All Playbook are in info mode verified.  
- [x] All Playbook are in inactive mode verified.

### CODE REVIEW

File: polyswarm/info.json [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    "fields": [
+      {
+        "name": "api_key",
+        "title": "API Key",
+        "type": "text",
+        "editable": true,
+        "visible": true,
+        "required": true,
+        "tooltip": "Specify the PolySwarm API Key",
+        "description": "Specify the PolySwarm API Key"
```
Comment (cs-omkar-kumbhar) [inline polyswarm/info.json]: check for better description

File: polyswarm/info.json [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+      "operation": "ip_reputation",
+      "title": "Scan IP",
+      "annotation": "ip_reputation",
+      "description": "Uploads a IP to PolySwarm and retrieves the analysis results.",
+      "category": "investigation",
+      "visible": true,
+      "enabled": true,
+      "parameters": [
+        {
+          "name": "url",
```
Comment (cs-omkar-kumbhar) [inline polyswarm/info.json]: the variable name can be changed to ip

File: polyswarm/info.json [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+      "operation": "domain_reputation",
+      "title": "Scan Domain",
+      "annotation": "domain_reputation",
+      "description": "Uploads a URL, IP, domain to PolySwarm and retrieves the analysis results.",
+      "category": "investigation",
+      "visible": true,
+      "enabled": true,
+      "parameters": [
+        {
+          "name": "url",
```
Comment (cs-omkar-kumbhar) [inline polyswarm/info.json]: the variable name can be changed to domain

File: polyswarm/operations.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        api.submit('https://polyswarm.io', artifact_type='url')
+        return True
+    except Exception as e:
+        raise ConnectorError("Invalid API Key")
+
+
+operations = {
+    'url_reputation': url_reputation,
+    'ip_reputation': url_reputation,
+    'domain_reputation': url_reputation,
```
Comment (cs-omkar-kumbhar) [inline polyswarm/operations.py]: change the function name as it is used for 3 operations, it is not url specific

File: polyswarm/info.json [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+          "title": "Domain",
+          "type": "text",
+          "editable": true,
+          "visible": true,
+          "required": true,
+          "tooltip": "The Domain to scan.",
+          "description": "Queries PolySwarm for Domain reputation information."
```
Comment (cs-omkar-kumbhar) [inline polyswarm/info.json]: check for better description

File: polyswarm/info.json [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+      "parameters": [
+        {
+          "name": "url",
+          "title": "IP",
+          "type": "text",
+          "editable": true,
+          "visible": true,
+          "required": true,
+          "tooltip": "The IP to scan.",
+          "description": "Queries PolySwarm for IP reputation information."
```
Comment (cs-omkar-kumbhar) [inline polyswarm/info.json]: check for better description

File: polyswarm/info.json [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+      "parameters": [
+        {
+          "name": "url",
+          "title": "URL",
+          "type": "text",
+          "editable": true,
+          "visible": true,
+          "required": true,
+          "tooltip": "The URL to scan.",
+          "description": "Queries PolySwarm for URL reputation information."
```
Comment (cs-omkar-kumbhar) [inline polyswarm/info.json]: check for better description, it should be like - Specify the URL for which you want to retrieve the reputation information.

File: polyswarm/info.json (hunk 1 above)
Comment (cs-utkarsh-surana) [reply polyswarm/info.json]: resolved

File: polyswarm/info.json (hunk 2 above)
Comment (cs-utkarsh-surana) [reply polyswarm/info.json]: changed to artifact

File: polyswarm/info.json (hunk 3 above)
Comment (cs-utkarsh-surana) [reply polyswarm/info.json]: changed to artifact

File: polyswarm/operations.py (hunk 4 above)
Comment (cs-utkarsh-surana) [reply polyswarm/operations.py]: done

File: polyswarm/info.json (hunk 5 above)
Comment (cs-utkarsh-surana) [reply polyswarm/info.json]: updated

File: polyswarm/info.json (hunk 6 above)
Comment (cs-utkarsh-surana) [reply polyswarm/info.json]: resolved

File: polyswarm/info.json (hunk 7 above)
Comment (cs-utkarsh-surana) [reply polyswarm/info.json]: done

**Review by cs-omkar-kumbhar — APPROVED** [review]

File: polyswarm/operations.py [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+                'params': {
+                    'hash': params.get('hash'),
+                },
+            },
+            result_parser=None,
+        ).execute()
+        response = result.raw_result.json()
+        for result in response.get('result'):
+            if result.get('failed'):
+                raise ConnectorError("Failed to get result.")
```
Comment (cs-prerna-joshi) [inline polyswarm/operations.py]: specify the response code and reason here.

File: polyswarm/info.json [hunk 9]
```diff
@@ -0,0 +1,608 @@
+{
+  "name": "polyswarm",
+  "label": "PolySwarm",
+  "version": "1.0.0",
+  "description": "PolySwarm is a real-time threat intelligence from a crowdsourced network of security experts and antivirus companies.",
```
Comment (cs-prerna-joshi) [inline polyswarm/info.json]:
```suggestion
  "description": "PolySwarm is a real-time threat intelligence from a crowdsourced network of security experts and antivirus companies.  This connector facilitates the automated operations to get the URL, IP, File, Domain reputation.",
```

File: polyswarm/info.json [hunk 10]
```diff
@@ -0,0 +1,608 @@
...
+        "type": "checkbox",
+        "name": "verify_ssl",
+        "value": true
+      }
+    ]
+  },
+  "operations": [
+    {
+      "operation": "get_url_reputation",
+      "title": "Get Url Reputation",
```
Comment (cs-prerna-joshi) [inline polyswarm/info.json]:
```suggestion
      "title": "Get URL Reputation",
```

File: polyswarm/operations.py [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+
+def wait_for_result(api, instance):
+    start = time.time()
+    timeout = settings.DEFAULT_SCAN_TIMEOUT
+    while True:
+        scan_result = core.PolyswarmRequest(
+            api,
+            {
+                'method': 'GET',
+                'url': '{}/consumer/submission/{}/{}'.format(api.uri, api.community, int(instance.artifact_id)),
```
Comment (cs-prerna-joshi) [inline polyswarm/operations.py]:
```suggestion
                'url': '{0}/consumer/submission/{1}/{2}'.format(api.uri, api.community, int(instance.artifact_id)),
```

File: polyswarm/connector.py [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+from connectors.core.connector import Connector, get_logger, ConnectorError
+from .operations import operations, _check_health
+logger = get_logger('polyswarm')
+
+
+class PolySwarm(Connector):
+    def execute(self, config, operation, params, **kwargs):
+        try:
+            config['connector_info'] = {"connector_name": self._info_json.get('name'),
+                "connector_version": self._info_json.get('version')}
```
Comment (cs-prerna-joshi) [inline polyswarm/connector.py]: This code is not required.

Comment (cs-utkarsh-surana) [reply polyswarm/connector.py]: removed

File: polyswarm/operations.py (hunk 11 above)
Comment (cs-utkarsh-surana) [reply polyswarm/operations.py]: updated

File: polyswarm/info.json (hunk 10 above)
Comment (cs-utkarsh-surana) [reply polyswarm/info.json]: updated

File: polyswarm/info.json (hunk 9 above)
Comment (cs-utkarsh-surana) [reply polyswarm/info.json]: updated

**Pull request merged**